src/seedu/addressbook/data/AddressBook.java

```diff
@@ -158,4 +158,13 @@ public int hashCode() {
         // use this method for custom fields hashing instead of implementing your own
         return Objects.hash(allPersons, allTags);
     }
+    /**
+     * Resets this addressbook to a certain state
+     * @param allPersons
+     * @param allTags
+     */
+    public void reset(UniquePersonList allPersons, UniqueTagList allTags){
+    	this.allPersons=allPersons;
+    	this.allTags=allTags;
+    }
```

Comment:
Actually, since `allTags` is just a set of all the tags all persons have, doesn't that mean that only `setAllPersons()` is needed?